src/chip.tsx

```diff
@@ -67,25 +81,60 @@ const Chip: React.FC<ChipProps> = (props: ChipProps) => {
                     <Icon color="currentColor" size={pxToRem(16)} />
                 </Box>
             )}
-            <Text2 id={id} medium truncate={1} color="currentColor">
-                {children}
-            </Text2>
+            {onClose || badge ? (
+                <Box paddingRight={8}>
+                    <Text2 id={id} medium truncate={1} color="currentColor">
+                        {children}
+                    </Text2>
+                </Box>
+            ) : (
+                <Text2 id={id} medium truncate={1} color="currentColor">
+                    {children}
+                </Text2>
+            )}
         </>
     );
 
-    const paddingLeft = Icon ? 8 : 12;
-
+    if (badge) {
+        return (
+            <Box
+                className={
+                    overAlternative ? styles.chipVariants.overAlternative : styles.chipVariants.default
+                }
+                paddingLeft={paddingLeft}
+                paddingRight={paddingIcon}
+                {...getPrefixedDataAttributes(dataAttributes, 'Chip')}
+            >
+                {body}
+                <Box>
+                    <IconButton
+                        size={18}
+                        style={{
+                            display: 'flex',
+                            justifyContent: 'center',
+                            alignItems: 'center',
+                            cursor: 'default',
+                        }}
+                        onPress={() => {}}
+                    >
```

Comment:
why IconButton? IconButton is meant to be used for clickable Icons. A badge isn't an icon an it isn't clickable either